docs/docs/howto/registry-authentication.md

```diff
@@ -49,8 +49,184 @@ EOF
 contrast generate "$@"
 ```
 
-### Bare metal
+### Confidential guest VMs
+
+For the [Contrast image puller](../architecture/components/runtime.md#contrast-image-puller) to be able to pull images from a private registry, it must be configured with the required authentication details.
+Contrast uses its own `TOML`-based configuration format for this purpose.
+This configuration must be provided as a secret[^1] to the [node installer `DaemonSet`](../architecture/components/runtime.md#node-installer-daemonset).
+The node installer then handles installing the secret in the runtime directory on the host, from where the runtime then forwards it to the confidential pod VM, allowing the image puller to read it.
+
+[^1]: Kubernetes secrets are accessible to anyone with API access to the node, as well as anyone with access to `etcd` ([source](https://kubernetes.io/docs/concepts/configuration/secret/)).
+    Anyone with these permissions is additionally able to change the configuration's content, with no way for Contrast to verify the integrity of the configuration eventually received by the image puller.
+
+    This allows an attacker to redirect image pull requests and configure an attacker-controlled registry to appear as legitimate.
+    The attacker could then, for example, log which images are being pulled.
+
+    This does **not** allow an attacker to serve malicious images or alter the contents of valid images, since irrespective of the image puller's configuration,
+    pulled images are integrity-protected through their mandatory pinned digest.
+
+:::warning
+
+Since the image puller configuration is annotated to the pod as an initdata document, it can be retrieved by any role with `get` or `list` permission for pods.
```

Comment:
```suggestion
Since the image puller configuration is annotated to the pod, it can be retrieved by any role with `get` or `list` permission for pods.
```

Let's not confuse this with initdata, which is integrity protected.